src/project/internal/notationproject.cpp

```diff
@@ -592,10 +593,21 @@ Ret NotationProject::doSave(const muse::io::path_t& path, engraving::MscIoMode i
         IF_ASSERT_FAILED(params.mode != MscIoMode::Unknown) {
             return make_ret(Ret::Code::InternalError);
         }
+        if (ioMode == MscIoMode::Zip
+            && globalConfiguration()->devModeEnabled()
+            && savePath.contains(" - CORRUPTED.mscz")
+            && !savePath.contains("autosave")) {
```

Comment:
We should do it differently:

1. add bool autosave to NotationProject::saveScore. This function is called from NotationProject::save where we already know the save mode. Use it here instead of checking the suffix

2. don't call checkSavedFileForCorruption if autosave == true